lib/naughty/core.lua

```diff
@@ -428,8 +428,11 @@ end
 -- @tparam[opt] func args.callback Function that will be called with all arguments.
 --   The notification will only be displayed if the function returns true.
 --   Note: this function is only relevant to notifications sent via dbus.
--- @tparam[opt] table args.actions Mapping that maps a string to a callback when this
---   action is selected.
+-- @tparam[opt] table args.actions Table describing all of the actions that are
+--   supported by the notification client. The table key is an action identifier.
+--   The table value is a record table indexed by two keys: `text` is a localized
+--   description string which is displayed to a user, and `callback` is a function
+--   called when this action is selected.
```

Comment:
The naughty [dbus handling code](https://github.com/awesomeWM/awesome/blob/master/lib/naughty/dbus.lua#L188) is the one who uses public API to display a received notification over DBus so this change is required to be able to pass three pieces (action-id, action-text, action-callback) instead of just two (action-id, action-callback). But it definitely seems (since you mention it) that the code in the core.lua doesn't care about action-id, which serves solely for the needs of the DBus dispatching.

So I guess now I can alter the PR to just pass the action nice description and callback in the old fashion `actions[text] = callback` table. I'm not sure if multiple actions with the same title make sense anyway. So let's wait some for the others and decide then.